OnFailedProcessMessage/handler.ts

```diff
@@ -23,21 +40,61 @@ type Handler = (c: Context, i: unknown) => Promise<void>;
  */
 export const getOnFailedProcessMessageHandler = ({
   lMessageStatusModel,
+  lMessageModel,
   telemetryClient
 }: IOnFailedProcessMessageHandlerInput): Handler =>
   withJsonInput(
-    withDecodedInput(CreatedMessageEvent, async (_, { messageId }) => {
-      await getMessageStatusUpdater(
-        lMessageStatusModel,
-        messageId
-      )(MessageStatusValueEnum.FAILED)();
-
-      telemetryClient.trackEvent({
-        name: "api.messages.create.failedprocessing",
-        properties: {
-          messageId
-        },
-        tagOverrides: { samplingEnabled: "false" }
-      });
-    })
+    withDecodedInput(CreatedMessageEvent, async (_, { messageId }) =>
+      pipe(
+        // Query for message with input messageId in order to retrieve the fiscalCode
+        lMessageModel.getQueryIterator({
+          parameters: [{ name: "@messageId", value: messageId }],
+          query: `SELECT * FROM m WHERE m.id = @messageId`
+        }),
+        flattenAsyncIterable,
+        asyncIterableToArray,
+        constant,
+        T.map(messages =>
+          messages.length === 1
+            ? pipe(messages[0], E.mapLeft(CosmosDecodingError))
+            : E.left(
+                CosmosErrorResponse({
+                  code: 404,
+                  message: "Missing message",
+                  name: "Not Found"
+                })
+              )
+        ),
+        TE.chainEitherKW(x =>
```

Comment:
The whole `chain` should be not needed. The `getQueryIterator` already have a decode applied to all the document retrieved from the database.